src/agentscope/service/web_search/web_digest.py

```diff
@@ -0,0 +1,273 @@
+        with `content` field is a dict, where keys are `html_parsing_types` and
+        values are the (pre-processed) parsed content. For example, if
+        `html_parsing_types = ['raw', 'selected_tags_to_text']` the
+            `content` field is
+            {
+                "raw": raw_content_of_web,
+                "selected_tags_to_text":
+                    processed_text_content_of_the_selected_tags,
+            }
+    """
+    header = {
+        "Accept-Language": "zh-CN,zh;q=0.9,en;q=0.8,en-GB;q=0.7,en-US;q=0.6",
+        "Cache-Control": "max-age=0",
+        "Upgrade-Insecure-Requests": "1",
+        "User-Agent": "Mozilla/5.0 (X11; Linux x86_64)"
+        " AppleWebKit/537.36 (KHTML, like Gecko) ",
+    }
+    try:
+        response = requests.get(url=url, headers=header, timeout=timeout)
+        if response.status_code == 200:
+            content_type = response.headers["Content-Type"].lower()
+            if "html" in content_type:
+                return parse_html(
+                    response.text,
+                    keep_raw,
+                    html_selected_tags,
+                    html_parse_func,
+                )
+            elif "pdf" in content_type:
+                # TODO: support pdf in the future
+                raise NotImplementedError(
+                    "Unsupported url to PDF content.",
+                )
+            elif "json" in content_type:
+                return ServiceResponse(
+                    ServiceExecStatus.SUCCESS,
+                    content=json.loads(response.text),
+                )
+            elif "image" in content_type:
+                # TODO: to support image (gif, jpeg, png) data
+                logger.warning(
+                    "Current implementation returns binary "
+                    "response.content for url with image Content-Types",
+                )
+                return ServiceResponse(
+                    ServiceExecStatus.SUCCESS,
+                    content=response.content,
+                )
+            else:
+                raise NotImplementedError(
+                    f"Unsupported content type ({content_type}) "
+                    f"with url: ({url})",
+                )
+        else:
+            logger.warning(
+                f"Fail to load web page, "
+                f"status code {response.status_code}",
+            )
+            return ServiceResponse(
+                ServiceExecStatus.ERROR,
+                content="",
+            )
+    except Exception as e:
+        logger.warning(e)
+        return ServiceResponse(ServiceExecStatus.ERROR, content="")
+
+
+def parse_html(
+    html_text: str,
+    keep_raw: bool = True,
+    html_selected_tags: Optional[Sequence[str]] = None,
+    html_parse_func: Optional[Callable] = None,
+) -> ServiceResponse:
+    """
+    Parse the obtained HTML file.
+    Args:
+        html_text (str):
+            HTTP response text
```

Comment:
"HTTP response text" maybe confused. Is it the HTML source code?

Reply:
fixed